src/ActionList/Item.tsx

```diff
@@ -192,17 +206,41 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
 
     const containerProps = _PrivateItemWrapper ? {role: role || itemRole ? 'none' : undefined} : menuItemProps
 
-    const wrapperProps = _PrivateItemWrapper ? menuItemProps : {}
+    const wrapperProps = _PrivateItemWrapper
+      ? menuItemProps
+      : {
+          sx: {
+            display: 'flex',
+            paddingX: isTopLevelInteractive() ? 0 : 2,
+            paddingY: isTopLevelInteractive() ? 0 : '6px',
+            flexGrow: 1,
+          },
+        }
 
     return (
       <ItemContext.Provider value={{variant, disabled, inlineDescriptionId, blockDescriptionId}}>
         <LiBox ref={forwardedRef} sx={merge<BetterSystemStyleObject>(styles, sxProp)} {...containerProps} {...props}>
+          {/* @ts-ignore onClick prop is only passed when _PrivateItemWrapper is set by ActionList.LinkItem. */}
           <ItemWrapper {...wrapperProps}>
             <Selection selected={selected} />
             {slots.leadingVisual}
             <Box
               data-component="ActionList.Item--DividerContainer"
-              sx={{display: 'flex', flexDirection: 'column', flexGrow: 1, minWidth: 0}}
+              sx={{
+                display: 'flex',
+                flexDirection: 'column',
+                flexGrow: 1,
+                minWidth: 0,
+                borderStyle: 'none',
+                backgroundColor: 'transparent',
+                cursor: 'inherit',
+                fontSize: 'inherit',
+                color: getVariantStyles(variant, disabled).color,
+                lineHeight: '20px',
+              }}
+              // @ts-ignore `as` prop may be passed to ActionList.Item, even if it isn't defined in ActionListItemProps.
+              // If this item is inside an ActionMenu, don't render an interactive button.
+              as={isTopLevelInteractive() ? 'div' : 'button'}
```

Comment:
https://primer-124f090a4d-13348165.drafts.github.io/storybook/?path=/story/components-actionlist--link-item-playground

Any `ActionList.LinkItem` is going to have an internal div instead of a button.

Reply:
`ActionMenu` is another example where we render a `div`. Each `<li>` has `role=menuitem`, which makes them interactive. If we had a `<button>` as the child, it breaks accessibility as we would have twice as many selectable elements. And the `<li>`s must be so since they're the direct children of a list element.

https://primer-124f090a4d-13348165.drafts.github.io/storybook/?path=/story/components-actionmenu--default

Reply:
Thank you for the examples I think I was assuming this element would become the listitem or the link -- not be the child.